### PR TITLE
Adds a Vector/BM25 example.

### PR DESCRIPTION
**Pull request by BillFarber**

### CODE REVIEW

File: examples/langchain/marklogic_bm25Vector_query_retriever.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            const result = fn.subsequence(cts.search(cts.andQuery([cts.wordQuery({}),cts.collectionQuery({})]),["score-bm25"]), 1, {});
+            let uris = [];
+            for(const doc of result){{
+                uris.push(xdmp.nodeUri(doc))
+            }}
+            const qv = vec.vector({})
+
+            const rows = op.fromView('{}','{}','')
+            .where(op.in(op.col('uri'), uris))
+            .bind(op.as("content", op.col("text")))
```
[the remark this answers is not in the document]
Comment (BillFarber) [reply examples/langchain/marklogic_bm25Vector_query_retriever.py:67]: We don't. Just an artifact from a simpler time.
Removed.

File: examples/langchain/README.md [hunk 2]
```diff
@@ -147,3 +149,18 @@ into different collections.
...
+
+### Running the Vector Query
+
+You are now ready to test the example vector retriever. Run the following to ask a
+question with the results augmented via the `marklogic_vector_query_retriever.py` module
+in this project:
+
+    python ask_vector_query.py "What is task decomposition?" posts_with_embeddings
+
+This retriever uses a CTS search for retrieving candidate documents, and defaults to
```
Comment (rjrudin) [inline examples/langchain/README.md:161]: Not important right now, but ultimately, I think we only want to say "This retriever searches MarkLogic for candidate documents" - i.e. some / many users will have no idea what "CTS search" means.

File: examples/langchain/marklogic_vector_query_retriever.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+class MarkLogicVectorQueryRetriever(BaseRetriever):
+
+    client: Client
+    embedding_generator: AzureOpenAIEmbeddings
+    max_results: int = 10
+    collections: List[str] = []
+    tde_schema: str
+    tde_view: str
+    scoring: str
```
Comment (rjrudin) [inline examples/langchain/marklogic_vector_query_retriever.py:16]: Calling it `scoring_method`, that appears to be the official term in our docs.

**Review by rjrudin — APPROVED** [review]
Couple tweaks, then merge away.

**Pull request merged**